source/lib.ts

```diff
@@ -76,18 +76,23 @@ const promisifyStore = (passedStore: LegacyStore | Store): Store => {
 	return new PromisifiedStore()
 }
 
+type OptionsWithLegacyStore = Omit<Partial<Options>, 'store'> & {
+	store?: Store | LegacyStore
+}
+
 /**
  * Type-checks and adds the defaults for options the user has not specified.
  *
  * @param options {Options} - The options the user specifies.
  *
  * @returns {Options} - A complete configuration object.
  */
-const parseOptions = (
-	passedOptions: Omit<Partial<Options>, 'store'> & {
-		store?: Store | LegacyStore
-	},
-): Options => {
+const parseOptions = (passedOptions: OptionsWithLegacyStore): Options => {
+	// Passing undefined should be equivalent to not passing an option at all, so we'll
+	// omit all fields where their value is undefined.
+	const omittedOptions: OptionsWithLegacyStore =
+		omitUndefinedOptions(passedOptions)
+
 	// See ./types.ts#Options for a detailed description of the options and their
 	// defaults.
 	const options = {
```

Comment:
With the new `Configuration` interface, this could be made:

```suggestion
	const config: Configuration = {
```